### ISSUE TITLE
[Arm64/Unix] Regression on tip

### ISSUE THREAD

**Issue by sdmaclea** [dotnet/coreclr#12524]
Tip has regressed and these test are failing on b05580db65032 
```
CoreMangLib/cti/system/collections/generic/queueenumerator/EnumeratorCurrent/EnumeratorCurrent.sh
CoreMangLib/cti/system/collections/generic/queueenumerator/EnumeratorDispose/EnumeratorDispose.sh
CoreMangLib/cti/system/collections/generic/queueenumerator/EnumeratorMoveNext/EnumeratorMoveNext.sh
CoreMangLib/cti/system/collections/generic/queue/QueueClear/QueueClear.sh
CoreMangLib/cti/system/collections/generic/queue/QueueContains/QueueContains.sh
CoreMangLib/cti/system/collections/generic/queue/QueueCopyTo/QueueCopyTo.sh
CoreMangLib/cti/system/collections/generic/queue/QueueCount/QueueCount.sh
CoreMangLib/cti/system/collections/generic/queue/QueueCtor2/QueueCtor2.sh
CoreMangLib/cti/system/collections/generic/queue/QueueCtor3/QueueCtor3.sh
CoreMangLib/cti/system/collections/generic/queue/QueueDequeue/QueueDequeue.sh
CoreMangLib/cti/system/collections/generic/queue/QueueEnqueue/QueueEnqueue.sh
CoreMangLib/cti/system/collections/generic/queue/QueueGetEnumerator/QueueGetEnumerator.sh
CoreMangLib/cti/system/collections/generic/queue/QueuePeek/QueuePeek.sh
CoreMangLib/cti/system/collections/generic/queue/QueueToArray/QueueToArray.sh
CoreMangLib/cti/system/collections/generic/stack/StackContains/StackContains.sh
CoreMangLib/system/collections/generic/hashset/Regression_Dev10_624201/Regression_Dev10_624201.sh
JIT/CheckProjects/CheckProjects/CheckProjects.sh
JIT/Performance/CodeQuality/BenchmarksGame/k-nucleotide/k-nucleotide/k-nucleotide.sh
JIT/Performance/CodeQuality/BenchmarksGame/regexdna/regexdna/regexdna.sh
JIT/Performance/CodeQuality/Linq/Linq/Linq.sh
JIT/Performance/CodeQuality/Roslyn/CscBench/CscBench.sh
JIT/Performance/CodeQuality/Serialization/Deserialize/Deserialize.sh
JIT/Performance/CodeQuality/Serialization/Serialize/Serialize.sh
JIT/Performance/CodeQuality/SIMD/RayTracer/RayTracer/RayTracer.sh
Loader/classloader/regressions/dev11_95728/dev11_95728/dev11_95728.sh
```

Typical failure signature is a seg fault or the following assert.
```
Assert failure(PID 40517 [0x00009e45], Thread: 40517 [0x9e45]): pArrayMT->IsArray()
    File: /home/vmjenkins/workspace/Dotnet/build_and_test/src/vm/jithelpers.cpp Line: 3156
    Image: /home/vmjenkins/workspace/Dotnet/build_and_test/bin/tests/pr1/Tests/coreoverlay/corerun
```

**Comment by sdmaclea** [comment]
@dotnet/arm64-contrib

**Comment by sdmaclea** [comment]
`ab7f3407fa9a0` was passing.  I will do a bisect between ab7f3407fa9a0 and b05580d

**Comment by sdmaclea** [comment]
Looks like 8cc7e35dd0a6 is the bad commit.  This is #12369 per the comments.

**Comment by sdmaclea** [comment]
@jkotas @briansull @ruben-ayrapetyan Looks like #12369 touched amd64 and arm assembly and stubs.  arm64 was not touched, was this because arm64 CI was never run?  Also I did not notice x86 changes, but perhaps they were included in amd64

**Comment by ruben-ayrapetyan** [comment]
@sdmaclea, there are two known issues related to this commit. The issues were fixed after b05580d - commits 91163768c4eec24379491e0bff49e8c4333f53f1 and 3951920cbd825c2b2251ba3572ca354f887d6033.

**Comment by sdmaclea** [comment]
@ruben-ayrapetyan Thanks.  I can confirm it is fixed for arm64 on 217ab30d021

**Comment by ruben-ayrapetyan** [comment]
@sdmaclea, Thank you.
